Looks like undoing this cycle exposed more code that was inadvertently depending on the dependency cycles (which is not surprising, I've been wading through a lot of this throughout the project). I'll look into fixing these to unblock you:

```
lib/libswiftSILGen.a(SILGen.cpp.o):SILGen.cpp:function swift::ASTLoweringRequest::evaluate(swift::Evaluator&, swift::ASTLoweringDescriptor) const: error: undefined reference to 'swift::SerializedSILLoader::getAllForModule(swift::Identifier, swift::FileUnit*)'
lib/libswiftSIL.a(SILModule.cpp.o):SILModule.cpp:function swift::SILModule::~SILModule(): error: undefined reference to 'swift::SerializedSILLoader::~SerializedSILLoader()'
lib/libswiftSIL.a(SILModule.cpp.o):SILModule.cpp:function swift::SILModule::lookUpDefaultWitnessTable(swift::ProtocolDecl const*, bool): error: undefined reference to 'swift::SerializedSILLoader::lookupDefaultWitnessTable(swift::SILDefaultWitnessTable*)'
lib/libswiftSIL.a(SILModule.cpp.o):SILModule.cpp:function swift::SILModule::getSILLoader(): error: undefined reference to 'swift::SerializedSILLoader::SerializedSILLoader(swift::ASTContext&, swift::SILModule*, swift::DeserializationNotificationHandlerSet*)'
lib/libswiftSIL.a(SILModule.cpp.o):SILModule.cpp:function swift::SILModule::getSILLoader(): error: undefined reference to 'swift::SerializedSILLoader::~SerializedSILLoader()'
lib/libswiftSIL.a(SILModule.cpp.o):SILModule.cpp:function swift::SILModule::deleteWitnessTable(swift::SILWitnessTable*): error: undefined reference to 'swift::SerializedSILLoader::invalidateWitnessTable(swift::SILWitnessTable*)'
lib/libswiftSIL.a(SILModule.cpp.o):SILModule.cpp:function swift::SILModule::loadFunction(swift::SILFunction*, swift::SILModule::LinkingMode): error: undefined reference to 'swift::SerializedSILLoader::lookupSILFunction(swift::SILFunction*, bool)'
lib/libswiftSIL.a(SILModule.cpp.o):SILModule.cpp:function swift::SILModule::loadFunction(llvm::StringRef, swift::SILModule::LinkingMode, llvm::Optional<swift::SILLinkage>): error: undefined reference to 'swift::SerializedSILLoader::lookupSILFunction(llvm::StringRef, llvm::Optional<swift::SILLinkage>)'
lib/libswiftSIL.a(SILModule.cpp.o):SILModule.cpp:function swift::SILModule::updateFunctionLinkage(swift::SILFunction*): error: undefined reference to 'swift::SerializedSILLoader::lookupSILFunction(swift::SILFunction*, bool)'
lib/libswiftSIL.a(SILModule.cpp.o):SILModule.cpp:function swift::SILModule::hasFunction(llvm::StringRef): error: undefined reference to 'swift::SerializedSILLoader::hasSILFunction(llvm::StringRef, llvm::Optional<swift::SILLinkage>)'
lib/libswiftSIL.a(SILModule.cpp.o):SILModule.cpp:function swift::SILModule::invalidateSILLoaderCaches(): error: undefined reference to 'swift::SerializedSILLoader::invalidateAllCaches()'
lib/libswiftSIL.a(SILModule.cpp.o):SILModule.cpp:function swift::SILModule::invalidateFunctionInSILCache(swift::SILFunction*): error: undefined reference to 'swift::SerializedSILLoader::invalidateFunction(swift::SILFunction*)'
lib/libswiftSIL.a(SILModule.cpp.o):SILModule.cpp:function swift::SILModule::eraseGlobalVariable(swift::SILGlobalVariable*): error: undefined reference to 'swift::SerializedSILLoader::invalidateGlobalVariable(swift::SILGlobalVariable*)'
lib/libswiftSIL.a(SILModule.cpp.o):SILModule.cpp:function swift::SILModule::lookUpVTable(swift::ClassDecl const*, bool): error: undefined reference to 'swift::SerializedSILLoader::lookupVTable(swift::ClassDecl const*)'
lib/libswiftSIL.a(SILModule.cpp.o):SILModule.cpp:function swift::SILModule::lookUpMoveOnlyDeinit(swift::NominalTypeDecl const*, bool): error: undefined reference to 'swift::SerializedSILLoader::lookupMoveOnlyDeinit(swift::NominalTypeDecl const*)'
lib/libswiftSIL.a(SILModule.cpp.o):SILModule.cpp:function swift::SILModule::loadDifferentiabilityWitness(swift::SILDifferentiabilityWitness*): error: undefined reference to 'swift::SerializedSILLoader::lookupDifferentiabilityWitness(swift::SILDifferentiabilityWitnessKey)'
lib/libswiftSIL.a(Linker.cpp.o):Linker.cpp:function swift::SILLinkerVisitor::deserializeAndPushToWorklist(swift::SILFunction*): error: undefined reference to 'swift::SerializedSILLoader::lookupSILFunction(swift::SILFunction*, bool)'
lib/libswiftSIL.a(Linker.cpp.o):Linker.cpp:function swift::SILLinkerVisitor::visitProtocolConformance(swift::ProtocolConformanceRef): error: undefined reference to 'swift::SerializedSILLoader::lookupWitnessTable(swift::SILWitnessTable*)'
```

--- a/lib/SIL/CMakeLists.txt
+++ b/lib/SIL/CMakeLists.txt
@@ -4,7 +4,9 @@ target_link_libraries(swiftSIL PUBLIC
   swiftDemangling)
 target_link_libraries(swiftSIL PRIVATE
   swiftAST
-  swiftClangImporter)
+  swiftClangImporter
+  swiftSema
+  swiftSerialization)
 
 add_subdirectory(IR)
 add_subdirectory(Utils)

--- a/lib/SILGen/CMakeLists.txt
+++ b/lib/SILGen/CMakeLists.txt
@@ -36,6 +36,7 @@ add_swift_host_library(swiftSILGen STATIC
   SILGenThunk.cpp
   SILGenType.cpp)
 target_link_libraries(swiftSILGen PRIVATE
+  swiftSerialization
   swiftSIL)
 
 set_swift_llvm_is_available(swiftSILGen)

--- a/tools/sil-passpipeline-dumper/CMakeLists.txt
+++ b/tools/sil-passpipeline-dumper/CMakeLists.txt
@@ -5,6 +5,7 @@ add_swift_host_tool(sil-passpipeline-dumper
 target_link_libraries(sil-passpipeline-dumper PRIVATE
   swiftFrontend
   swiftIRGen
+  swiftSema
   swiftSILGen
   swiftSILOptimizer
   # Clang libraries included to appease the linker on linux.

